src/server_manager/web_app/gcp_account.ts

```diff
@@ -90,19 +74,102 @@ export class GcpAccount implements gcp.Account {
     return result;
   }
 
+  /** @see {@link Account#listLocations}. */
+  async listLocations(projectId: string): Promise<gcp.RegionMap> {
+    const listZonesResponse = await this.apiClient.listZones(projectId);
+    const zones = listZonesResponse.items ?? [];
+
+    const result: gcp.RegionMap = {};
+    zones.map((zone) => {
+      const region = zone.region.substring(zone.region.lastIndexOf('/') + 1);
+      if (!(region in result)) {
+        result[region] = [];
+      }
+      // TODO: Check status
+      if (zone.status) {
+        result[region].push(zone.name);
+      }
+    });
+    return result;
+  }
+
+  /** @see {@link Account#listProjects}. */
+  async listProjects(): Promise<Project[]> {
+    const response = await this.apiClient.listProjects();
+    if (response.projects?.length > 0) {
+      return response.projects.map(project => {
+        return {
+          id: project.projectId,
+          name: project.name,
+        };
+      });
+    }
+    return [];
+  }
+
+  // TODO: Add API call error handling.
+  /** @see {@link Account#createProject}. */
+  async createProject(id: string, billingAccountId: string): Promise<Project> {
```

Comment:
The code should be resilient to failures and be recoverable in case of incomplete creation.
This means you will need methodd to check the project status and break down the logic.

See my comments on the UI.

Reply:
I think we can still have it as a single function, but you need to check the status before doing the action.

Also, the UI will need methods to check if the project is ready, and to fetch the project id and billing account.